sql/api/query.go

```diff
@@ -14,6 +14,7 @@ type Query struct {
 	Query        string `json:"query"`
 	// Deprecated: Use databricks_job resource to schedule a Query
 	Schedule       *QuerySchedule    `json:"schedule"`
+	RunAsRole      string            `json:"run_as_role,omitempty"`
```

Comment:
If the API fills `run_as_role` with default `owner`, then it makes sense to add `suppress_diff`, like this: https://github.com/databricks/terraform-provider-databricks/blob/master/sql/resource_sql_dashboard.go#L19

Reply:
Hi Alex, can you explain what the `suppress_diff` exactly does? I've read the documentation in the [Contributor docs](https://github.com/databricks/terraform-provider-databricks/blob/master/CONTRIBUTING.md#adding-a-new-resource), but it's not quite clear to me yet.

Reply:
It's suppressing the diff from a default value to an empty string.  It's needed because if you declare the resource without `run_as_role`, then API will fill it with default `owner`, but when you run apply next time, because `run_as_role` is empty in the query, then it will be detected as drift, and will try to reset it back.

Reply:
Ah, makes sense. However, `suppress_diff` is already defined on the [QueryEntity](https://github.com/KOR-Financial/terraform-provider-databricks/blob/fix/2939_sql_query_run_as_role/sql/resource_sql_query.go#L27)'s RunAsState property. Isn't the Query api object you're referencing here used to represent the Databricks API objects, so unrelated to the Terraform state, and therefore requiring no `suppress_diff` config here?

Reply:
ah, sorry - misread it